File: ShrutiVarshney_BanasthaliVidyapith_StringAssignment/StringAssignmentQuestions/src/stringassignmentquestions/Problem2.java
package stringassignmentquestions;
import java.util.*;
public class Problem2 
{
    public static void main(String[] args) 
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter a string containing binary numbers only:");
        String inputStr = sc.next();
        int minPackets = findMinPackets(inputStr);
        System.out.println(minPackets);
    }

    public static int findMinPackets(String str) 
    {
        int count = 0;
        for (int i = 1; i < str.length(); i++) 
        {
            if (str.charAt(i) != str.charAt(i - 1)) 
            {
                count++;
            }
        }
        return count + 1;
    }
}

File: ShaliniKumari_BanasthaliVidyapith/Arrays/String/Problem2.cpp
#include <iostream>
#include <string>
using namespace std;

int findMinPackets(string str) {
    int count = 0;
    for (int i = 1; i < str.length(); i++) {
        if (str[i] != str[i - 1]) {
            count++;
        }
    }
    return count + 1;
}

int main() {
    string inputStr;
    cout << "Enter a string containing binary numbers only: ";
    cin >> inputStr;

    int minPackets = findMinPackets(inputStr);
    cout << minPackets << endl;

    return 0;
}
